test/hotspot/jtreg/serviceability/dcmd/vm/ClassLoaderHierarchyTest.java

```diff
@@ -60,39 +73,77 @@ public class ClassLoaderHierarchyTest {
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_2_cl);
+
+        // C) Test output for several class loaders, same class, same name, empty parents,
+        //    and all these should be folded by default.
+        EmptyDelegatingLoader emptyLoader3 = new EmptyDelegatingLoader("EmptyLoader3", null);
+        EmptyDelegatingLoader emptyLoader4 = new EmptyDelegatingLoader("EmptyLoader4", emptyLoader3);
+        ClassLoader named_child_3_cl = new TestClassLoader("ChildX", emptyLoader4); // Same names
+        ClassLoader named_child_4_cl = new TestClassLoader("ChildX", emptyLoader4);
+        ClassLoader named_child_5_cl = new TestClassLoader("ChildX", emptyLoader4);
+        ClassLoader named_child_6_cl = new TestClassLoader("ChildX", emptyLoader4);
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_3_cl);
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_4_cl);
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_5_cl);
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_6_cl);
+
+        // D) Test output for several *unnamed* class loaders, same class, same parents,
+        //    and all these should be folded by default too.
+        EmptyDelegatingLoader emptyLoader5 = new EmptyDelegatingLoader(null, null);
+        EmptyDelegatingLoader emptyLoader6 = new EmptyDelegatingLoader(null, emptyLoader5);
+        ClassLoader named_child_7_cl = new TestClassLoader(null, emptyLoader6); // Same names
+        ClassLoader named_child_8_cl = new TestClassLoader(null, emptyLoader6);
+        ClassLoader named_child_9_cl = new TestClassLoader(null, emptyLoader6);
+        ClassLoader named_child_10_cl = new TestClassLoader(null, emptyLoader6);
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_7_cl);
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_8_cl);
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_9_cl);
+        loadTestClassInLoaderAndCheck("TestClass2", named_child_10_cl);
 
         // First test: simple output, no classes displayed
         OutputAnalyzer output = executor.execute("VM.classloaders");
-        output.shouldContain("<bootstrap>");
-        output.shouldMatch(".*TestClassLoader");
-        output.shouldMatch("Kevin.*TestClassLoader");
-        output.shouldMatch("Bill.*TestClassLoader");
+        // (A)
+        output.shouldContain("+-- <bootstrap>");
+        output.shouldContain("      +-- \"platform\", jdk.internal.loader.ClassLoaders$PlatformClassLoader");
+        output.shouldContain("      |     +-- \"app\", jdk.internal.loader.ClassLoaders$AppClassLoader");
+        output.shouldContain("      +-- \"Kevin\", ClassLoaderHierarchyTest$TestClassLoader");
+        output.shouldContain("      +-- ClassLoaderHierarchyTest$TestClassLoader");
+        output.shouldContain("      |     +-- \"Bill\", ClassLoaderHierarchyTest$TestClassLoader");
```

Comment:
Can you test using one long string, and do the same for each section below? I think embedding \\R as the newline character should work on all platforms. If you run into unforeseen issues, then don't worry about.

Reply:
Fixed